### PR TITLE
ZCS-1528: Configure Ephemeral Data Backend URL via Admin UI

### PR DESCRIPTION
**Pull request by haitran1696**

### CODE REVIEW

File: WebRoot/js/zimbraAdmin/globalconfig/controller/ZaGlobalConfigViewController.js [hunk 1]
```diff
@@ -377,6 +377,73 @@ function () {
...
+	if (ZaItem.hasWritePermission(ZaGlobalConfig.A_zimbraEphemeralBackendURL, tmpObj)) {
+		if (tmpObj.attrs[ZaGlobalConfig.A_zimbraEphemeralBackendURL]) {
+			var backendURL = tmpObj.attrs[ZaGlobalConfig.A_zimbraEphemeralBackendURL];
+			var chunks = backendURL.split(":");
+			
+			if (chunks < 2 || chunks > 3) {
+				throw new AjxException(AjxMessageFormat.format(ZaMsg.ERROR_INVALID_BACKEND_URL, [backendURL]), AjxException.INVALID_PARAM, "ZaGlobalConfigViewController.prototype._saveChanges");
+			}
+
+			//check input backend with currently supported backends (ldap, ssdb)
```
Comment (grishick) [inline WebRoot/js/zimbraAdmin/globalconfig/controller/ZaGlobalConfigViewController.js]: lets remove this validation. Our SSDB backend implementation was intended as default and reference implementation, but admins should be free to implement their own server extensions for other backends and use them.

File: WebRoot/js/zimbraAdmin/globalconfig/controller/ZaGlobalConfigViewController.js [hunk 2]
```diff
@@ -419,6 +486,26 @@ function () {
 	return true;
 }
 
+ZaGlobalConfigViewController.isValidHostName =
+function (isHostNameStr) {
+	if (!isHostNameStr) {
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/globalconfig/controller/ZaGlobalConfigViewController.js]: minor: indentation

File: WebRoot/js/zimbraAdmin/globalconfig/controller/ZaGlobalConfigViewController.js [hunk 3]
```diff
@@ -419,6 +486,26 @@ function () {
...
+        return false;
+    }
+    
+    //Invalid host name that has two or more adjacent dot, dot is at index 0 or at the end of host name
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/globalconfig/controller/ZaGlobalConfigViewController.js]: same as above

File: WebRoot/js/zimbraAdmin/globalconfig/controller/ZaGlobalConfigViewController.js (hunk 2 above)
Comment (haitran1696) [reply WebRoot/js/zimbraAdmin/globalconfig/controller/ZaGlobalConfigViewController.js]: I see line of code "if (!isHostNameStr) {"  is indented. Please review again or tell me what need to be changed at that line!

**Pull request merged**